test/config-luatest/app_test.lua

```diff
@@ -0,0 +1,110 @@
+local t = require('luatest')
+local helpers = require('test.config-luatest.helpers')
+
+local g = helpers.group()
+
+-- Start a script that is pointed by app.file or app.module.
+--
+-- TODO: Verify that the script has access to config:get() values.
+g.test_startup_success = function(g)
```

Comment:
I think it's a little odd that the test is added here, but logically it should be added in the next commit.

Reply:
Are you about all the new cases? If so, they work without the second patch. And, I think, they're useful without it: it verifies the logic of the `app` section with running of a new instance with a config and without using internals.

Or about these two commented lines (and TODO comments)? If so, I'm OK to remove them and add in the next commit (instead of uncommenting).